### PR TITLE
[EN-1388] Moves consul role to own repository

### PR DESCRIPTION
**Pull request by agirivera**
Story: http://jira.juwai.com/browse/EN-1388
Sub-task: http://jira.juwai.com/browse/EN-1397
- Adds galaxy tags
- Fixes download url of consul packages, had moved to https://releases.hashicorp.com
- No logical changes on the role from _ansible-playbooks_

How to test:
1. `git clone -b EN-1388 https://github.com/agirivera/ansible-role-consul.git`
2. `cd ansible-role-consul/tests`
3. `ansible-galaxy install -p roles -r requirements.yml`. _python27_ and _supervisor_ roles will fail for the moment, run `git clone -b EN-1388 https://github.com/agirivera/ansible-role-python27.git roles/juwai.python27` and `git clone -b EN-1388 https://github.com/agirivera/ansible-role-supervisor.git roles/juwai.supervisor` instead
4. `vagrant up`

/cc @juwai/platform @tjoelsson @imlazyone Please review. Thanks.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+consul_server_name: "{{ inventory_hostname }}"
+consul_node_name: "{{ inventory_hostname }}"
+
+rpc_services:
+  - {
+    name: time_service,
+    port: 8081,
+    tags: "['rpc']",
+    check: "zerorpc --connect tcp://127.0.0.1:8081 --timeout 1 _zerorpc_ping",
```
Comment (agirivera) [inline defaults/main.yml:51]: @tjoelsson Does it makes sense for us to make `zerorpc-python` a dependency of this role? I'm not sure yet how these hooks are actually installed to make an educated decision.

Comment (tjoelsson) [inline defaults/main.yml:51]: Yes, I think that would make sense. We need the command line tool to poll the services.

**Comment by tjoelsson** [comment]
The playbook works, but consul doesn't start. Here is error from logfile:

`==> Error decoding '/home/consul/consul_0.5.0/consul.d/rpc_services.json': invalid character '\'' looking for beginning of value`

**Comment by agirivera** [comment]
@tjoelsson Fixed config error.

**Comment by tjoelsson** [comment]
I provisioned a new box for consul-alerts with the latest updates from this role. The playbook finishes, but checking the process status in the box I see this:

```
consul                               FATAL     Exited too quickly (process log may have details)
consul_alerts:consul_alerts_daemon   FATAL     Exited too quickly (process log may have details)
consul_alerts:consul_watch           RUNNING   pid 22034, uptime 0:02:00
```

**Comment by tjoelsson** [comment]
Here is what's written to the log:

```
==> WARNING: It is highly recommended to set GOMAXPROCS higher than 1
==> Starting Consul agent...
==> Starting Consul agent RPC...
==> Joining -wan cluster...
==> Must be a server to join WAN cluster
```

**Comment by agirivera** [comment]
@tjoelsson: Fixed and updated. `consul` server starts properly and as expected now. Please review.

File: defaults/main.yml [hunk 2]
```diff
@@ -26,16 +26,16 @@ json_false: "false"
 # config values from http://www.consul.io/docs/agent/options.html
 consul_datacenter: dc1
 consul_client_address: 0.0.0.0
-consul_advertise_address: "{{ ansible_ssh_host }}"
-consul_bind_address: "{{ ansible_ssh_host }}"
+consul_advertise_address: "{{ ansible_host }}"
+consul_bind_address: "{{ ansible_host }}"
```
Comment (tjoelsson) [inline defaults/main.yml]: ? I guess this makes it 2.0+ only

File: README.md [hunk 3]
```diff
@@ -13,33 +13,33 @@ Role Variables
 
 Available variables are listed below, along with default values (see `defaults/main.yml`):
 
-## server
+## consul_server
 
 Install consul on server or not.
 
-Default is `false`.
+Default is `true`.
```
Comment (tjoelsson) [inline README.md:20]: I don't quite understand this setting. When should it be false?

Comment (agirivera) [inline README.md:20]: Not really sure myself. But from what I've gather after looking back at `ansible-playbooks`, some hosts redefine the setting to true to provision it completely. I'm thinking other hosts that don't really need the service leave it as false but still uses some of the variables defined in the role.

File: defaults/main.yml (hunk 2 above)
Comment (agirivera) [inline defaults/main.yml]: Ah.. I didn't think about that at all. I'll revert these.

**Comment by tjoelsson** [comment]
It works now 👍

**Comment by agirivera** [comment]
Awesome :smile_cat: @imlazyone Looks good?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+------------
+
++ juwai.supervisor, when supervisor_enabled
+
+Example Playbook
+----------------
+
+    - hosts: servers
+      roles:
+        - { role: consul, when: server}
```
Comment (stuartwan) [inline README.md]: ? juwai.consul?

File: meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  # this branch. If travis integration is cofigured, only notification for this
+  # branch will be accepted. Otherwise, in all cases, the repo's default branch
+  # (usually master) will be used.
+  #github_branch:
+
+  platforms:
+  - name: EL
+    versions:
+    - 6
+  - name: Amazon
```
Comment (stuartwan) [inline meta/main.yml]: ? comment out this for now?

File: README.md (hunk 4 above)
Comment (agirivera) [inline README.md]: Done.

File: meta/main.yml (hunk 5 above)
Comment (agirivera) [inline meta/main.yml]: Done. Good catch :smile_cat:

**Comment by stuartwan** [comment]
🚢

**Comment by agirivera** [comment]
@tjoelsson Still ok?

**Comment by tjoelsson** [comment]
Still good 👍

**Pull request merged**